### ISSUE TITLE
Create Developer-Mode Setting (related to SIWE domain binding)

### ISSUE THREAD

**Issue by digiwand** [MetaMask/metamask-extension#18531]
### Description

In order to protect users, we want to strictly enforce domain-binding logic for Sign-in With Ethereum (SIWE) while supporting the developers. 

To do this, we'd like a developer-mode setting that will allow developers to disable the strict domain-binding check. 

**See details and discussion here:**
https://github.com/MetaMask/metamask-extension/issues/18188#issuecomment-1497120809

### Prerequisite
Needs design

### Spec
- Needs to be communicated to developers
- Clearly state developer-mode is unsafe 
- Unlock developer-mode only in Flask or beta
  - Is there a way to redirect users from stable to Flask or beta if it may be applicable? ideas: note in settings, add in response when RPC requests are rejected due to domain binding. 
- Unsafe developer mode unblocks domain bindings
  - To confirm, will this bypass warning modals as well? 

----

Another consideration has been to apply a specific setting to disable the domain-binding check in Flask or beta. This could be a plan B.
Proposed here: https://github.com/MetaMask/metamask-extension/issues/18188#issuecomment-1499919118

**Comment by digiwand** [comment]
Duplicate of https://github.com/MetaMask/metamask-extension/issues/18531
